### PR TITLE
[TTAHUB-1189] Fix deleting issues

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Fixes a bug with goal info not being cleared out when the last goal is removed. This also explains the loading screen (attempting to fetch related info for the selected goal)

## How to test
To repro bug on main-ar-redesign
1) Add a goal to a new AR
2) Save draft
3) Save goal
4) Remove that goal, so that there are none left
You will see the form partially filled out, and there will be a loading screen

This PR fixes that issue

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1189

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by kryswisnaskas** [comment]
Looks good! I verified that there is no filled out form under a loader when removing a goal.

I did find though that the deleted goal comes back.
- remove the goal
- refresh the page
- goal is back

File: frontend/src/pages/ActivityReport/Pages/goalsObjectives.js [hunk 1]
```diff
@@ -116,11 +116,16 @@ const GoalsObjectives = ({
...
+
     onUpdateGoals(copyOfSelectedGoals);
 
     // if we have no goals, open the form up via the
     // hander provided by the context
     if (copyOfSelectedGoals.length === 0) {
+      setValue('goalForEditing', '');
+      setValue('goalName', '');
+      setValue('goalEndDate', '');
+      setValue('goalIsRttapa', '');
```
Comment (kryswisnaskas) [inline frontend/src/pages/ActivityReport/Pages/goalsObjectives.js:128]: Can we also add/update the test?

**Comment by thewatermethod** [comment]
> I did find though that the deleted goal comes back.
>    remove the goal
>    refresh the page
>    goal is back

Isn't this expected with no save?

File: frontend/src/pages/ActivityReport/Pages/goalsObjectives.js (hunk 1 above)
Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/goalsObjectives.js:128]: https://github.com/HHS/Head-Start-TTADP/pull/1163/commits/7ea3c48c0280f4de424140bf7e2b855060dfa51d

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good just verify anything outstanding questions with Krys.

**Comment by thewatermethod** [comment]
I'm going to merge this - if we decide we want the delete to trigger a save, we can handle that in a separate PR

**Pull request merged**